pkg/experiment/symbolizer/symbolizer.go

```diff
@@ -0,0 +1,844 @@
+		lidiaData, ei, err = s.processELFData(data)
+		if err != nil {
+			level.Error(s.logger).Log(
+				"msg", "symbolizer: Failed to process ELF data",
+				"build_id", req.BuildID,
+				"error", err,
+			)
+			return err
+		}
+	}
+
+	// Create a reader for the Lidia table
+	lidiaReader := &memoryReader{
+		bs:  lidiaData,
+		off: 0,
+	}
+
+	// Open the lidia table from memory
+	table, err := lidia.OpenReader(lidiaReader, lidia.WithCRC())
+	if err != nil {
+		s.metrics.debugSymbolResolution.WithLabelValues("lidia_error").Observe(0)
+		level.Error(s.logger).Log("msg", "opening Lidia table", "err", err)
+		return fmt.Errorf("open lidia file: %w", err)
+	}
+	defer table.Close()
+
+	// If we're reading a Lidia file, get the binary layout from the table
+	if isLidia {
+		lidiaLayout, err := table.GetBinaryLayout()
+		if err != nil {
+			level.Error(s.logger).Log(
+				"msg", "failed to get binary layout from Lidia table",
+				"build_id", req.BuildID,
+				"error", err,
+			)
+			return err
+		}
+		ei = convertBinaryLayout(lidiaLayout)
+	}
+
+	return s.symbolizeWithTable(ctx, table, ei, req)
+}
+
+func (s *ProfileSymbolizer) storeLidiaTableInCache(data []byte, buildID string, cacheStart time.Time) error {
+	var lidiaData []byte
+	var ei *BinaryLayout
+	var err error
+
+	// Check if data is already in Lidia format by looking for the magic number
+	if len(data) >= 4 && data[0] == 0x2e && data[1] == 0x64 && data[2] == 0x69 && data[3] == 0x61 {
+		// Data is already in Lidia format
+		lidiaData = data
+
+		// Create a reader for the Lidia table
+		lidiaReader := &memoryReader{
+			bs:  lidiaData,
+			off: 0,
+		}
+
+		// Open the Lidia table to extract binary layout
+		table, err := lidia.OpenReader(lidiaReader, lidia.WithCRC())
+		if err != nil {
+			return fmt.Errorf("open lidia file: %w", err)
+		}
+		defer table.Close()
+
+		// Get binary layout from the Lidia table
+		lidiaLayout, err := table.GetBinaryLayout()
+		if err != nil {
+			return fmt.Errorf("get binary layout from lidia table: %w", err)
+		}
+
+		// Convert to our internal binary layout format
+		ei = convertBinaryLayout(lidiaLayout)
+	} else {
+		// Process as ELF data
+		lidiaData, ei, err = s.processELFData(data)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Store both the processed Lidia table data and the BinaryLayout in the cache
+	entry := LidiaTableCacheEntry{
+		Data: lidiaData,
+		EI:   ei,
+	}
+
+	success := s.lidiaTableCache.Set(buildID, entry, int64(len(lidiaData)))
+	s.metrics.cacheOperations.WithLabelValues("debuginfo_cache", "set", StatusSuccess).Observe(time.Since(cacheStart).Seconds())
+	if !success {
+		level.Warn(s.logger).Log("msg", "failed to store debug info in cache", "buildID", buildID, "size", len(lidiaData))
+	}
+	s.metrics.cacheSizeBytes.WithLabelValues("debuginfo_cache").Set(float64(len(lidiaData)))
+
+	return nil
+}
+
+func (s *ProfileSymbolizer) processELFData(data []byte) (lidiaData []byte, ei *BinaryLayout, err error) {
+	// Create a reader from the data
```

Comment:
Lets remove all the "obvious" comments all over the new package. They are not very useful.

> // Create a reader from data
> ... bytes.NewReader
> 
> // Create lidia file
> lidia.CreateLidiaFromElf
> 
> 		// Convert to our internal binary layout format
>		ei = convertBinaryLayout(lidiaLayout)
> 
> // Open the Lidia table to extract binary layout
> lidia.Open